internal/helm/controller/controller.go

```diff
@@ -93,6 +104,21 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 	return nil
 }
 
+// parsePredicateSelector parses the selector in the WatchOptions and creates a predicate
+// that is used to filter resources based on the specified selector
+func parsePredicateSelector(selector metav1.LabelSelector) ctrlpredicate.Predicate {
+	// If a selector has been specified in watches.yaml, add it to the watch's predicates.
+	if !reflect.ValueOf(selector).IsZero() {
+		p, err := ctrlpredicate.LabelSelectorPredicate(selector)
+		if err != nil {
+			log.Error(err, "Unable to create predicate from selector.")
```

Comment:
```suggestion
			return fmt.Errorf("error constructing predicate from watches selector: %v", err), nil
```

Reply:
do you mean `return nil, fmt.Errorf("error constructing predicate from watches selector: %v", err)` since the first return type is predicate and the second return type is error?

Reply:
Yes sorry.